pkg/asset/installconfig/kubevirt/validation.go

```diff
@@ -0,0 +1,119 @@
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// Validate executes kubevirt specific validation
+func Validate(ic *types.InstallConfig, clientBuilder ClientBuilderFuncType) error {
+	kubevirtPlatformPath := field.NewPath("platform", "kubevirt")
+
+	if ic.Platform.Kubevirt == nil {
+		return errors.New(field.Required(
+			kubevirtPlatformPath,
+			"validation requires a Engine platform configuration").Error())
+	}
+
+	return validatePlatform(ic.Platform.Kubevirt, ic.MachineNetwork, kubevirtPlatformPath, clientBuilder).ToAggregate()
+}
+
+func validatePlatform(kubevirtPlatform *kubevirt.Platform, machineNetworkEntryList []types.MachineNetworkEntry, fldPath *field.Path, clientBuilder ClientBuilderFuncType) field.ErrorList {
+	allErrs := validation.ValidatePlatform(kubevirtPlatform, fldPath)
+	ctx := context.Background()
+
+	client, resultErrs := validateInfraClusterReachable(ctx, kubevirtPlatform.Namespace, fldPath, clientBuilder)
+	allErrs = append(allErrs, resultErrs...)
+	if client != nil {
+		allErrs = append(allErrs, validateStorageClassExistsInInfraCluster(ctx, kubevirtPlatform.StorageClass, client, fldPath)...)
+		allErrs = append(allErrs, validateNetworkAttachmentDefinitionExistsInInfraCluster(ctx, kubevirtPlatform.NetworkName, kubevirtPlatform.Namespace, client, fldPath)...)
+	}
+	allErrs = append(allErrs, validateIPsInMachineNetworkEntryList(machineNetworkEntryList, kubevirtPlatform.APIVIP, kubevirtPlatform.IngressVIP, fldPath)...)
+
+	return allErrs
+}
+
+// validateInfraClusterReachable validates the following:
+// 1. Client can be created -  login to cluster is done or KUBECONFIG environment variable was exported
+// 2. Cluster is reachable - valid user
+// 3. Namespace exists
+// 4. Kubevirt installed
+// 5. User can list VMs in namespace
+func validateInfraClusterReachable(ctx context.Context, namespace string, fieldPath *field.Path, clientBuilder ClientBuilderFuncType) (Client, field.ErrorList) {
+	allErrs := field.ErrorList{}
+	client, err := clientBuilder()
+	if err != nil {
+		detailedErr := fmt.Errorf("failed to create InfraCluster client with error: %v", err)
+		allErrs = append(allErrs, field.Invalid(fieldPath.Child("InfraClusterReachable"), "InfraCluster", detailedErr.Error()))
+
+		return nil, allErrs
+	}
+
+	if _, err := client.ListVirtualMachine(namespace); err != nil {
+		detailedErr := fmt.Errorf("failed to access to InfraCluster with error: %v", err)
+		allErrs = append(allErrs, field.Invalid(fieldPath.Child("InfraClusterReachable"), "InfraCluster", detailedErr.Error()))
+
+		return nil, allErrs
+	}
+
+	return client, allErrs
+}
+
+func validateStorageClassExistsInInfraCluster(ctx context.Context, name string, client Client, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if _, err := client.GetStorageClass(ctx, name); err != nil {
+		detailedErr := fmt.Errorf("failed to get storageClass %s from InfraCluster, with error: %v", name, err)
```

Comment:
s/storageClass/StorageClass/

also, InfraCluster needs to be something more friendly and all message should use the same form.

Reply:
Fixed